go/vt/vtgate/planbuilder/symboltable.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package planbuilder
+
+import (
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// SymbolTable contains the symbols for a SELECT
+// statement. If it's for a subquery, it points to
+// an outer scope.
+type SymbolTable struct {
+	tables map[sqlparser.SQLName]*TableAlias
+	outer  *SymbolTable
+}
+
+// TableAlias is part of SymbolTable.
+// It represnts a table alias in a FROM clause.
+type TableAlias struct {
+	// Name represents the name of the alias.
+	Name sqlparser.SQLName
+	// Keyspace points to the keyspace to which this
+	// alias belongs.
+	Keyspace *Keyspace
```

Comment:
What if you have a subquery that spans multiple keyspaces? (I didn’t see an answer to this question)?